### PR TITLE
Update 3.4

### PR DESCRIPTION
**Pull request by psteyer**
idna version update from `3.3` -> `3.4`

### CODE REVIEW

File: recipe/meta.yaml
Comment (skupr-anaconda) [inline recipe/meta.yaml]: Should be `flit_core >=3.2,<4`, see https://github.com/kjd/idna/blob/44fa5a0a7f7e3a5051b4d3e2ae7957c0494cd708/pyproject.toml#L2

File: recipe/meta.yaml
Comment (skupr-anaconda) [inline recipe/meta.yaml:23]: `setuptools` is no more needed as a build backend, see https://github.com/kjd/idna/blob/44fa5a0a7f7e3a5051b4d3e2ae7957c0494cd708/pyproject.toml#L3
But the maintainers still use it for legacy users. So we can add a comment for `setuptools`:
```
  # setuptools is needed for legacy users, see https://github.com/kjd/idna/commit/9234d29b38e56e863db03af5980819c031221725
  - setuptools
```

File: recipe/meta.yaml
Comment (skupr-anaconda) [inline recipe/meta.yaml:28]: Please remove `python <3.10` from `test/requires `to unblock the tests for **python 3.10**. It was a temporary fix.

File: recipe/meta.yaml
Comment (psteyer) [reply recipe/meta.yaml]: Added the pinning

File: recipe/meta.yaml
Comment (psteyer) [reply recipe/meta.yaml:23]: Added comment

File: recipe/meta.yaml
Comment (psteyer) [reply recipe/meta.yaml:28]: Removed the python from `test/requires`.

I was curious, but where did you find this?

**Review by skupr-anaconda — APPROVED** [review]

File: recipe/meta.yaml
Comment (skupr-anaconda) [reply recipe/meta.yaml]: Sorry for misleading you, the name should be `flit-core` (with a dash) not `flit_core`.

**Review by danpetry — APPROVED** [review]

**Review by varlackc — APPROVED** [review]
LGTM

File: recipe/meta.yaml
Comment (skupr-anaconda) [reply recipe/meta.yaml:28]: There was an issue with python 3.10 linking by `conda` when **3.10** was interpreted as **3.1**. That's why we skip testing on that moment.

**Pull request merged**
